components/camel-jetty9/src/main/java/org/apache/camel/component/jetty9/JettyContentExchange9.java

```diff
@@ -214,14 +220,28 @@ public void onFailure(Request request, Throwable failure) {
             }
 
         };
-        BufferingResponseListener responseListener = new BufferingResponseListener() {
+
+        InputStreamResponseListener responseListener = new InputStreamResponseListener() {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+            @Override
+            public void onContent(Response response, ByteBuffer content, Callback callback) {
+                if (response.getStatus() != SC_OK) {
+                    LOG.warn("Response received {}: {}", response.getStatus(), response.getReason());
+                }
+                byte[] buffer = new byte[content.limit()];
+                content.get(buffer);
+                baos.write(buffer, 0, buffer.length);
+
+                callback.succeeded();
+            }
 
             @Override
             public void onComplete(Result result) {
                 if (result.isFailed()) {
                     doTaskCompleted(result.getFailure());
                 } else {
-                    onResponseComplete(result, getContent(), getMediaType());
+                    onResponseComplete(result, baos.toByteArray(), null);
```

Comment:
Is there a reason for null as media Type? Just curiosity :-)

Thanks, I will test and merge